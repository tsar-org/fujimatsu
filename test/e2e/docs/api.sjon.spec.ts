import {
	createExecutionContext,
	env,
	waitOnExecutionContext,
} from 'cloudflare:test';
import worker from '@/../src';
import { describe, expect, it } from 'vitest';

const IncomingRequest = Request<unknown, IncomingRequestCfProperties>;

describe('test GET /docs/api.json', () => {
	it('snapshot test', async () => {
		// act
		const request = new IncomingRequest('http://localhost:8787/docs/api.json');
		const ctx = createExecutionContext();
		const response = await worker.fetch(request, env, ctx);
		await waitOnExecutionContext(ctx);

		// assert
		expect(await response.status).toBe(200);
		expect(await response.json()).toMatchSnapshot();
	});
});
