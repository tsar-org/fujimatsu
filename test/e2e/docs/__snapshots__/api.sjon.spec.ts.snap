// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test GET /docs/api.json > snapshot test 1`] = `
{
  "components": {
    "parameters": {},
    "schemas": {},
  },
  "info": {
    "title": "Fujimatsu API Reference",
    "version": "1.0.0",
  },
  "openapi": "3.1.0",
  "paths": {
    "/": {
      "get": {
        "description": "Health check",
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "properties": {
                    "message": {
                      "type": "string",
                    },
                  },
                  "required": [
                    "message",
                  ],
                  "type": "object",
                },
              },
            },
            "description": "OK",
          },
        },
      },
    },
    "/v1/discord/oauth2/authorize": {
      "get": {
        "description": "Authorize",
        "responses": {
          "302": {
            "description": "Redirect to Discord Login URL",
          },
          "500": {
            "content": {
              "application/json": {
                "schema": {
                  "properties": {
                    "message": {
                      "enum": [
                        "Internal server error",
                      ],
                      "type": "string",
                    },
                  },
                  "required": [
                    "message",
                  ],
                  "type": "object",
                },
              },
            },
            "description": "Internal Server Error",
          },
        },
      },
    },
  },
}
`;
